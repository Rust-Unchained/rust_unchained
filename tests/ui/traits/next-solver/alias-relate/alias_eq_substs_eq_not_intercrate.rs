//@ compile-flags: -Znext-solver

// check that a `alias-eq(<?a as TraitB>::Assoc, <?b as TraitB>::Assoc)` goal fails
// during coherence. We must not incorrectly constrain `?a` and `?b` to be
// equal.

trait TraitB {
    type Assoc;
}

trait Overlaps<T> {}

impl<T: TraitB> Overlaps<Box<T>> for <T as TraitB>::Assoc {}
impl<U: TraitB> Overlaps<U> for <U as TraitB>::Assoc {}
//~^ ERROR conflicting implementations of trait

// The two bellow are additional requirements for failure in Unchained Rust
// Unchained Rust only forbids these impls if a conflict actually occurs.
impl TraitB for Box<i32> {
    type Assoc = i32;
}

impl TraitB for i32 {
    type Assoc = i32;
}

fn main() {}
