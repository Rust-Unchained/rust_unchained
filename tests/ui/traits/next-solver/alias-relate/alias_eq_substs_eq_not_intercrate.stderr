error[E0119]: conflicting implementations of trait `Overlaps<Box<i32>>` for type `i32`
  --> $DIR/alias_eq_substs_eq_not_intercrate.rs:14:1
   |
LL | impl<T: TraitB> Overlaps<Box<T>> for <T as TraitB>::Assoc {}
   | --------------------------------------------------------- first implementation here
LL | impl<U: TraitB> Overlaps<U> for <U as TraitB>::Assoc {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `i32`

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0119`.
